executor/partition_table.go

```diff
@@ -75,6 +86,11 @@ type nextPartitionForIndexMerge struct {
 func (n nextPartitionForIndexMerge) nextPartition(ctx context.Context, tbl table.PhysicalTable) (Executor, error) {
 	exec := n.exec
 	exec.table = tbl
+	for i := 0; i < len(exec.dagPBs); i++ {
```

Comment:
There's no need to update the dag for index merge. Because it's impossible to build an index merge with TiFlash.